algorithms/a3c/main.py

```diff
@@ -40,80 +54,276 @@
+parser.add_argument('--living-reward', type=float, default=0,
+                    help="""Default reward at each time step (default: 0,
+                    change to -0.005 to encourage shorter paths)""")
+parser.add_argument('--frame-width', type=int, default=300,
+                    help='Frame width (default: 300)')
+parser.add_argument('--frame-height', type=int, default=168,
+                    help='Frame height (default: 168)')
+parser.add_argument('-v', '--visualize', type=int, default=0,
+                    help="""Visualize the envrionment (default: 0,
+                    use 0 for faster training)""")
+parser.add_argument('--sleep', type=float, default=0,
+                    help="""Sleep between frames for better
+                    visualization (default: 0)""")
+parser.add_argument('--vizdoom', dest='vizdoom', action='store_true',
+                    help='Run vizdoom env instead with name below instead of ai2thor and atari')
+parser.add_argument('--scenario-path', type=str, default="vizdoom_maps/room.wad",
+                    help="""Doom scenario file to load
+                    (default: maps/room.wad)""")
+parser.add_argument('--interactive', type=int, default=0,
+                    help="""Interactive mode enables human to play
+                    (default: 0)""")
+parser.add_argument('--all-instr-file', type=str,
+                    default="vizdoom_data/instructions_all.json",
+                    help="""All instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_all.json)""")
+parser.add_argument('--train-instr-file', type=str,
+                    default="vizdoom_data/instructions_train.json",
+                    help="""Train instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_train.json)""")
+parser.add_argument('--test-instr-file', type=str,
+                    default="vizdoom_data/instructions_test.json",
+                    help="""Test instructions file path relative to a3c folder
+                    (default: vizdoom_data/instructions_test.json)""")
+parser.add_argument('--object-size-file', type=str,
+                    default="vizdoom_data/object_sizes.txt",
+                    help='Object size file path relative to a3c folder '
+                         '(default: data/object_sizes.txt)')
+parser.add_argument('-e', '--evaluate', type=int, default=0,
+                    help="""0:Train, 1:Evaluate MultiTask Generalization
+                    2:Evaluate Zero-shot Generalization (default: 0)
+                    async must be on and will only run test function""")
 
 if __name__ == '__main__':
-    os.environ['OMP_NUM_THREADS'] = '1'
     os.environ['CUDA_VISIBLE_DEVICES'] = ""
 
+    # todo add to README.md more info
+
     args = parser.parse_args()
+    # set to 0 so that checkpoint resume can overwrite if necessary
+    args.episode_number = 0
```

Comment:
You have a parameter to load the checkpoint so use it although I couldn't find it. I suggest something on these lines
```suggestion
    args.episode_number = args.checkpoint['episode_numer'] if args.checkpoint else 0
```
Also please add a docstring explaining what should I do to load an experiment and/or modify it. Right now it is all confusing to me

Reply:
I'm actually loading in the default value as 0 and then later on after reading in checkpoint (this requires a lot of logic to check if you even have a checkpoint) I overwrite